frontend/src/views/SecretMainPage/components/SecretListView/SecretReferenceDetails.tsx

```diff
@@ -0,0 +1,124 @@
+import { SecretV3RawSanitized, TSecretReferenceTraceNode } from "@app/hooks/api/types";
+
+type Props = {
+  environment: string;
+  secretPath: string;
+  secret: SecretV3RawSanitized;
+};
+
+export const SecretReferenceNode = ({
+  node,
+  isRoot
+}: {
+  node: TSecretReferenceTraceNode;
+  isRoot?: boolean;
+}) => (
+  <div>
+    {isRoot ? (
+      <FormControl label="Reference value" className="sticky top-0 mb-0 bg-mineshaft-800">
+        <SecretInput
+          key="value-overriden"
+          isReadOnly
+          value={node.value}
+          containerClassName="text-bunker-300 hover:border-primary-400/50 border border-mineshaft-600 bg-bunker-800  px-2 py-1.5"
+        />
+      </FormControl>
+    ) : (
+      <div className="py-2 text-sm">
+        <div>
+          {node.key}
+          <Tooltip content={node.value}>
+            <span
+              className={twMerge(
+                "ml-2 rounded border border-gray-400 px-1 text-xs text-gray-400",
+                !node.value && "border-red-400 text-red-400"
+              )}
+            >
+              {node.value ? "value" : "empty"}
+            </span>
+          </Tooltip>
+        </div>
+        <div className="mt-2">
+          <div className="flex items-center space-x-2">
+            <div className="flex items-center">
+              <Tag
+                size="xs"
+                className="mr-0 flex items-center rounded-r-none border border-mineshaft-500"
+              >
+                <div>Environment</div>
+              </Tag>
+              <Tag
+                size="xs"
+                className="flex items-center rounded-l-none border border-mineshaft-500 bg-mineshaft-900 pl-1"
+              >
+                <div className="max-w-[150px] overflow-hidden text-ellipsis whitespace-nowrap">
+                  {node.environment}
+                </div>
+              </Tag>
+            </div>
+            <div className="flex items-center">
+              <Tag
+                size="xs"
+                className="mr-0 flex items-center rounded-r-none border border-mineshaft-500"
+              >
+                <div>Secret Path</div>
+              </Tag>
+              <Tag
+                size="xs"
+                className="flex items-center rounded-l-none border border-mineshaft-500 bg-mineshaft-900 pl-1"
+              >
+                <div className="max-w-[150px] overflow-hidden text-ellipsis whitespace-nowrap">
+                  {node.secretPath}
+                </div>
+              </Tag>
+            </div>
+          </div>
+        </div>
+        <div className="w-full border-b border-gray-800 py-1" />
+      </div>
+    )}
+    {node.children.length > 0 && (
+      <div className="border-l border-gray-600 pl-6 transition-all hover:border-primary-600">
+        {node.children.map((el, index) => (
+          <SecretReferenceNode node={el} key={`node-${node.key}-${index + 1}`} />
+        ))}
+      </div>
+    )}
```

Comment:
I think children should be behind an expandable, so only the root references are shown initially giving the user the ability to expand each descendant instead of all being "expanded" initially

Reply:
I tried this but didn't like the experience. 

First most of references won't be in so many levels. So I don;t think there is much to expand. Second the space is already a bit smaller so i don't want to make interactions inside there